OpenRA.Game/Map/MapPreview.cs

```diff
@@ -174,7 +174,20 @@ public InnerData Clone()
 		readonly ModData modData;
 
 		public readonly string Uid;
-		public IReadOnlyPackage Package { get; private set; }
+		public string PackageName { get; private set; }
+		private IReadOnlyPackage _Package = null;
+		public IReadOnlyPackage Package {
+			get {
+				if (_Package == null)
+				{
+					_Package = parentPackage.OpenPackage(PackageName, modData.ModFiles);
+				}
+				return _Package;
+			}
+			private set {
+				_Package = value;
+			}
+		}
```

Comment:
This structure doesn't seem fully correct. MapDirectoryTracker cycles through all packages so it will re-create them. There are also a few places that want just the name of the package but now will open the package

Reply:
This is the "core" of the change. 
Packages are only loaded on-demand and then stay loaded if not explicitly unloaded using PackageDispose().
So in the first loadmaps loop I explicitly unload them by calling the PackageDispose function, but then if later the app code try to access them again then yes they will be reloaded. But I guess that's exactly what we want ?

Reply:
I see, but they were already created when there isn't a need. I just made them unloaded after first mapload.
So as I see the tracker is using Package.Name, wich will reload the package. 
I can use the new  PackageName field from MapPreview instead of the one in the Package to prevent it from loading the Package again ?

Reply:
I searched for .Package in the whole source code and fix this one and another one. Should be fine ? An alternative would be to create a getPackage() method than give the loaded package without keeping the reference but that would break the MapPreview class behavior because it is using Package internally to implement IReadOnlyFileSystem interface. Looks like a good compromise ?

Reply:
<img width="1165" alt="Screenshot 2024-04-24 at 17 16 35" src="https://github.com/OpenRA/OpenRA/assets/37534529/61c0dfaa-22bc-40cd-b4a5-1f4292be85e1">

there's still a few more. One in MissionBrowser, it would load all packages on entering the menu. Another in SaveMapLogic, it would do the same.